wen-restart/src/wen_restart.rs

```diff
@@ -480,4 +521,116 @@ mod tests {
+        };
+        let last_vote_slot: Slot = (RestartLastVotedForkSlots::MAX_SLOTS + expected_slots)
+            .try_into()
+            .unwrap();
+        let last_vote_bankhash = Hash::new_unique();
+        let last_voted_fork_slots_progress = WenRestartProgress {
+            state: RestartState::LastVotedForkSlots.into(),
+            my_last_voted_fork_slots: Some(LastVotedForkSlotsRecord {
+                last_vote_fork_slots: vec![last_vote_slot],
+                last_vote_bankhash: last_vote_bankhash.to_string(),
+                shred_version: shred_version as u32,
+            }),
+            last_voted_fork_slots_aggregate: None,
+        };
+        for (progress, last_vote, setup, expected_err, cleanup) in vec![
+            (
+                init_progress.clone(),
+                VoteTransaction::from(VoteStateUpdate::from(vec![(0, 8), (last_vote_slot, 1)])),
+                None,
+                "last vote is not a vote transaction",
+                None,
+            ),
+            (
+                init_progress,
+                VoteTransaction::from(Vote::new(vec![], Hash::new_unique())),
+                None,
+                "last vote does not have last voted slot",
+                None,
+            ),
+            (
+                last_voted_fork_slots_progress,
+                VoteTransaction::from(Vote::new(vec![last_vote_slot], last_vote_bankhash)),
+                Some(|| change_proto_file_readonly(&wen_restart_proto_path, true)),
+                "Permission denied",
+                Some(|| change_proto_file_readonly(&wen_restart_proto_path, false)),
+            ),
+        ] {
+            assert!(write_wen_restart_records(&wen_restart_proto_path, &progress,).is_ok());
+            if let Some(setup_function) = setup {
+                setup_function();
+            }
+            assert_matches!(wait_for_wen_restart(
+                &wen_restart_proto_path,
+                last_vote,
+                blockstore.clone(),
+                cluster_info.clone(),
+                bank_forks.clone(),
+                wen_restart_repair_slots.clone(),
+                80,
+                Arc::new(AtomicBool::new(false)),
+            ).err().unwrap().to_string(), message if message.contains(expected_err));
+            if let Some(cleanup_function) = cleanup {
+                cleanup_function();
+            }
+        }
+    }
+
+    #[test]
+    fn test_increment_and_write_wen_restart_records() {
+        let mut progress = WenRestartProgress {
+            state: RestartState::Init.into(),
+            my_last_voted_fork_slots: None,
+            last_voted_fork_slots_aggregate: None,
+        };
+        let my_dir = TempDir::new().unwrap();
+        let mut wen_restart_proto_path = my_dir.path().to_path_buf();
+        wen_restart_proto_path.push("wen_restart_status.proto");
+        for expected_state in [
+            RestartState::LastVotedForkSlots,
```

Comment:
Testing the initial state is `RestartState::Init` would be nice as well

Reply:
Done.